### PR TITLE
Feature/id 23 radar

### PR DESCRIPTION
**Pull request by karnokg**

### CODE REVIEW

File: src/AutomatedCar/SystemComponents/Sensors/Radar.cs [hunk 1]
```diff
@@ -4,27 +4,84 @@
...
+
+            var carPos = new Vector2(car.X, car.Y);
+
+            bool isInTriangle = this.PointInTriangle(
+                new Vector2(obj.X, obj.Y),
+                carPos,
+                carPos + this.RotatedVisionLeftPos,
+                carPos + this.RotatedVisionRightPos);
+
+            return obj is Car && isInTriangle;
```
Comment (erosd99) [inline src/AutomatedCar/SystemComponents/Sensors/Radar.cs:84]: 1. Vészféknek valszeg hirtelen lelépő gyalogosokra is kellene fékeznie, lehet azt is megéri nézni.
2. Itt nem lesz elég az isInTriangle, azokat az objecteket is nézni kell, ahol egy részük a szenzor látókörében van, de a középpontja még nincs.

File: src/AutomatedCar/SystemComponents/Sensors/Sensor.cs [hunk 2]
```diff
@@ -31,13 +32,65 @@ public Sensor(VirtualFunctionBus virtualFunctionBus)
             this.triangle = new SensorTriangle();
         }
 
+        protected abstract List<WorldObject> FilterRelevantWorldObjects();
+
+//        protected abstract void SaveWorldObjectsToPacket(List<WorldObject> worldObjects);
         protected abstract void SaveWorldObjectsToPacket();
 
-        protected abstract List<WorldObject> FilterRelevantWorldObjects();
+        protected AutomatedCar GetAutomatedCar()
```
Comment (erosd99) [inline src/AutomatedCar/SystemComponents/Sensors/Sensor.cs:40]: Ezt Egonnál is átbeszéltük, sztem ő már át is írta az implementációban, de nem kapjuk meg direktben a kocsit. Maga a kocsi a VFB-re írja fel a jelenlegi koordinátáit, és azt olvassa ki a szenzor.

File: src/AutomatedCar/SystemComponents/Sensors/Sensor.cs [hunk 3]
```diff
@@ -31,13 +32,65 @@ public Sensor(VirtualFunctionBus virtualFunctionBus)
...
+
+        /// <summary>
+        /// Returns whether point is in a triangle with Barycentric method.
+        /// </summary>
+        /// <param name="p">Point</param>
+        /// <param name="p0">Triangle Point 1.</param>
+        /// <param name="p1">Triangle Point 2.</param>
+        /// <param name="p2">Triangle Point 3.</param>
+        /// <returns>Point is in triangle or not.</returns>
+        protected bool PointInTriangle(Vector2 p, Vector2 p0, Vector2 p1, Vector2 p2)
```
Comment (erosd99) [inline src/AutomatedCar/SystemComponents/Sensors/Sensor.cs:82]: Ez a fv. lesz CollisionDetectionben. Emiatt blokkolva vagy annak mergeléséig. 

Amúgy szép, átlátható kód!

**Review by erosd99 — CHANGES_REQUESTED** [review]

File: src/AutomatedCar/SystemComponents/Sensors/Sensor.cs (hunk 2 above)
Comment (karnokg) [reply src/AutomatedCar/SystemComponents/Sensors/Sensor.cs:40]: A WorldObjecteket is a World singletonból kérjük le a szenzorokban, ami tartalmazza az AutomatedCart is (tehát onnan is el lehet érni az autót), akkor van bármi előnye a VFB-re packetet írni a koordinátáknak? Úgy érzem, hogy ez felesleges bonyolítás.

File: src/AutomatedCar/Helpers/CollisionDetection.cs [hunk 4]
```diff
@@ -185,5 +208,20 @@ public static PolylineGeometry RotateBoundingBox(PolylineGeometry boundingBox, d
         {
             return new PolylineGeometry(boundingBox.Points.Select(point => RotatePoint(point, angle)).ToList(), false);
         }
+
+        public static PolylineGeometry TransformGeometry(WorldObject obj, double angle = 0)
+        {
```
Comment (erosd99) [inline src/AutomatedCar/Helpers/CollisionDetection.cs:213]: Ebből minek kell másik? Konkrétan felette van ugyanaz a method rendes névvel, kommentálva.

File: src/AutomatedCar/Helpers/CollisionDetection.cs [hunk 5]
```diff
@@ -165,7 +166,29 @@ public static bool PointInTriangle(Point point, Tuple<Point, Point, Point> trian
...
-        private static Point RotatePoint(Point point, double angle)
+        /// <summary>
+        /// Returns whether point is in a triangle with Barycentric method.
+        /// </summary>
+        /// <param name="p">Point</param>
+        /// <param name="p0">Triangle Point 1.</param>
+        /// <param name="p1">Triangle Point 2.</param>
+        /// <param name="p2">Triangle Point 3.</param>
+        /// <returns>Point is in triangle or not.</returns>
+        public static bool PointInTriangle(Point p, Point p0, Point p1, Point p2)
```
Comment (erosd99) [inline src/AutomatedCar/Helpers/CollisionDetection.cs:177]: Ne legyen kettő PointInTriangle metódus. Ha ez fixálja a másik hibáit, akkor:
1. Töröld ki másikat
2. A háromszög egy Point Tuple legyen, részben backward compatibility miatt, részben azért, mert a Test Suite arra a függvényre van írva.
3. Summaryban kommentáld edge caseket.

**Review by erosd99 — CHANGES_REQUESTED** [review]

File: src/AutomatedCar/Helpers/CollisionDetection.cs (hunk 5 above)
Comment (erosd99) [reply src/AutomatedCar/Helpers/CollisionDetection.cs:177]: És légyszi pullolj developról kövi commit előtt, Collision Detection 1 verzióval előrébb van, mint a tied.

Comment (karnokg) [reply src/AutomatedCar/Helpers/CollisionDetection.cs:177]: Javítom, 10 perc, viszont ez azelőtt íródott, mielőtt az a pull request el lett fogadva.

**Review by erosd99 — APPROVED** [review]
LGTM
